File: .agent/scripts/verify_local_yaml.py
import os
import re
import yaml

def check_dir(dir_path):
    if not os.path.exists(dir_path):
        return
    print(f"Scanning {dir_path}...")
    error_count = 0
    for root, _, files in os.walk(dir_path):
        for file in files:
            if file.endswith(".md"):
                filepath = os.path.join(root, file)
                try:
                    with open(filepath, 'r', encoding='utf-8') as f:
                        content = f.read()
                    
                    match = re.match(r'^---\n(.*?)\n---', content, re.DOTALL)
                    if match:
                        yaml_str = match.group(1)
                        try:
                            # strict parse
                            yaml.safe_load(yaml_str)
                        except yaml.YAMLError as exc:
                            print(f"\n[X] CRITICAL YAML ERROR in {filepath}:")
                            print(exc)
                            error_count += 1
                except Exception as e:
                    print(f"Could not read {filepath}: {e}")
    if error_count == 0:
        print(f"All good in {dir_path}!")

check_dir("/home/terwu01/github/marffet/.agent/workflows")
check_dir("/home/terwu01/github/marffet/.agent/skills")

File: .agents/scripts/verify_local_yaml.mjs
import fs from 'fs';
import path from 'path';
import yaml from 'js-yaml';

function checkDir(dirPath) {
    if (!fs.existsSync(dirPath)) return;
    
    console.log(`Scanning ${dirPath}...`);
    let errorCount = 0;
    
    function walkSync(currentDirPath) {
        fs.readdirSync(currentDirPath).forEach(function(name) {
            var filePath = path.join(currentDirPath, name);
            var stat = fs.statSync(filePath);
            if (stat.isFile() && filePath.endsWith('.md')) {
                try {
                    const content = fs.readFileSync(filePath, 'utf8');
                    const match = content.match(/^---\n([\s\S]*?)\n---/);
                    if (match) {
                        const yamlStr = match[1];
                        try {
                            yaml.load(yamlStr);
                        } catch (e) {
                            console.log(`\n[X] CRITICAL YAML ERROR in ${filePath}:`);
                            console.log(e.message);
                            errorCount++;
                        }
                    }
                } catch (e) {
                    console.log(`Could not read ${filePath}: ${e.message}`);
                }
            } else if (stat.isDirectory()) {
                walkSync(filePath);
            }
        });
    }
    
    walkSync(dirPath);
    if (errorCount === 0) {
        console.log(`All good in ${dirPath}!`);
    }
}

checkDir('/home/terwu01/github/marffet/.agent/workflows');
checkDir('/home/terwu01/github/marffet/.agent/skills');
